packages/react/src/BlockNoteView.tsx

```diff
@@ -1,15 +1,52 @@
 import { BlockNoteEditor, BlockSchema } from "@blocknote/core";
+import { MantineProvider } from "@mantine/core";
 import { EditorContent } from "@tiptap/react";
-// import { BlockNoteTheme } from "./BlockNoteTheme";
-// import { MantineProvider } from "@mantine/core";
+import { getBlockNoteTheme } from "./BlockNoteTheme";
+import { ReactNode, useEffect, useState } from "react";
+import { FormattingToolbarPositioner } from "./FormattingToolbar/components/FormattingToolbarPositioner";
+import { HyperlinkToolbarPositioner } from "./HyperlinkToolbar/components/HyperlinkToolbarPositioner";
+import { SlashMenuPositioner } from "./SlashMenu/components/SlashMenuPositioner";
+import { SideMenuPositioner } from "./SideMenu/components/SideMenuPositioner";
 
 export function BlockNoteView<BSchema extends BlockSchema>(props: {
-  editor: BlockNoteEditor<BSchema> | null;
+  editor: BlockNoteEditor<BSchema>;
+  children?: ReactNode;
 }) {
+  const [ready, setReady] = useState(false);
+
+  useEffect(() => {
```

Comment:
Actually it's necessary due to the bug in TipTap which fires event listeners before the editor is ready. But the ready check should be moved to the listener hooks instead of the `BlockNoteView` either way.